File: ddd-briscola-web/src/main/ts/model/input.ts
module Model.Input {

  export type MatchKind = string | NumberOfGamesMatchKind | TargetPointsMatchKind
  
  export interface NumberOfGamesMatchKind {
    numberOfMatches:number
  }
  
  export interface TargetPointsMatchKind {
    winnerPoints:number
  }
  
  export type CompetitionStartDeadline = String | OnPlayerCountDeadline

  export interface OnPlayerCountDeadline {
    count:number
    kind:string
  }
  
  export interface Player {
    name:string
    password:string  
  }
  
  export interface Competition {
    players:Model.Ws.Path[]
    kind:MatchKind,
    deadline:CompetitionStartDeadline
  }

  export interface Card {
    seed:string
    "number":number  
  }

}

File: ddd-briscola-web/src/main/scala/org/obl/briscola/presentation/input.scala
package org.obl.briscola.presentation

import org.obl.briscola.player.GamePlayers
import org.obl.raz.Path

object Input {

  import org.obl.briscola.competition.MatchKind
  import org.obl.briscola.competition.CompetitionStartDeadline

  final case class Competition(players: GamePlayers, kind: MatchKind, deadline: CompetitionStartDeadline)
  final case class Player(name: String, password: String)

  /*
   * actually not used must: replace uses of corresponding models from org.obl.briscola.player
   */
//  sealed trait GamePlayers
//  final case class Players(players: Set[Path]) extends GamePlayers
//  final case class TeamPlayers(players: Set[TeamPlayer], teams:Set[TeamInfo]) extends GamePlayers
//  
//  final case class TeamPlayer(player:Path, teamName:String)
//  final case class TeamInfo(name:String)
//  
//  final case class Team(name:String, players:Set[Path])
//  final case class Teams(teams:Seq[Team]) 

  
}
 